### DISCUSSION TITLE
Not Able to clear all Asynchronous storage while app install in IOS

### DISCUSSION THREAD

**Discussion by Bharanitharan06** [react-native-async-storage/async-storage#624] [Q&A]
Hi, Am developing react native Application for **Android** and **IOS** and am using ** AsyncStorage.** for store some data. If user install and login with my app then session will created. if User uninstall and again he install my app I want to delete he's old session and want to create the new session but AsyncStorage. data is not cleared its maintained with old data.

In Android I fixed with **allowBackup** in **AndroidMainfest.xml** file. but **IOS** I don't know how to fix. give some ideas for it.

**Comment by tido64** [comment]
Are you looking for how to disable iCloud backup: https://react-native-async-storage.github.io/async-storage/docs/advanced/backup?

**Reply by Bharanitharan06** [reply]
no, but I want to clear my AsyncStorage while installing my app.I tried to add it in my info.plist and run my app but its not worked...`<key>UIViewControllerBasedStatusBarAppearance</key>
	<false/>
    <key>RCTAsyncStorageExcludeFromBackup</key>
    <false/>
	`

**Reply by krizzu** [reply]
With latest version, excluding from automatic backup is disabled, so the content of storage is removed when you delete the app.

**Reply by Bharanitharan06** [reply]
But not deleted in my case , I just want to clear all the AsyncStorage data while install the App in iOS

**Reply by krizzu** [reply]
You said you added `<key>RCTAsyncStorageExcludeFromBackup</key> <false/>`, but that tells iOS do **perform** the backup. By default, we turn this **off**.

I think what you see is that those files are already backed up, so _probably_ you'll have to delete the backup from the iCloud and retest it.
